app/lib/pre_assembly/batch.rb

```diff
@@ -107,7 +107,7 @@ def pre_assembled_object_containers
     def containers_via_manifest(&block)
       return enum_for(:containers_via_manifest) { manifest_rows.size } unless block_given?
 
-      manifest_rows.each_with_index do |manifest_row, i|
+      manifest_rows.each.with_index do |manifest_row, i|
```

Comment:
Unrelated to changes in this PR, but I wonder whether `i` is 0-indexed and, if so, whether that might be confusing in the error message below. (If I see there's a missing object in row 1, I might be surprised to see it in what I'd consider to be row 2.)

Reply:
I think this is a good question;  I'll make a separate ticket for it.